gis/gccview/gcc_layer_puller.py

```diff
@@ -0,0 +1,277 @@
+             "returnM": "false",
+             "orderByFields": "OBJECTID", 
+             "returnDistinctValues": "false",
+             "returnExtentsOnly": "false",
+             "resultOffset": "{}".format(max_number),
+             "resultRecordCount": "{}".format(record_max),
+             "f":"json"}
+    
+    for retry in range(3):
+        try:
+            r = requests.get(base_url, params = query, verify = False, timeout = 300)
+            r.raise_for_status()
+        except requests.exceptions.HTTPError as err_h:
+            LOGGER.error("Invalid HTTP response: ", err_h)
+        except requests.exceptions.ConnectionError as err_c:
+            LOGGER.error("Network problem: ", err_c)
+            sleep(10)
+        except requests.exceptions.Timeout as err_t:
+            LOGGER.error("Timeout: ", err_t)
+        except requests.exceptions.RequestException as err:
+            LOGGER.error("Error: ", err)
+        else:
+            return_json = r.json()
+            break
+    
+    return return_json
+
+def insert_data(output_table, insert_column, return_json, schema_name, con):
+    """
+    Function to insert data to our postgresql database, the data is inserted into a temp table (for audited tables)
+
+    Parameters
+    ----------
+    output_table : string
+        Table name for postgresql, returned from function get_tablename
+
+    insert_column : SQL composed
+        Composed object of column name and types use for creating a new postgresql table
+
+    return_json : json
+        Resulted json response from calling the api, returned from function get_data
+    
+    schema_name : string
+        The schema in which the table will be inserted into
+    
+    con: Airflow Connection
+        Could be the connection to bigdata or to on-prem server
+    """
+    rows = []
+    features = return_json['features']
+    fields = return_json['fields']
+    trials = [[field['name'], field['type']] for field in fields]
+
+    for feature in features:
+        geom = feature['geometry']
+        geometry_type = return_json['geometryType']
+        geometry = get_geometry(geometry_type, geom)
+        
+        row = []
+        for trial in trials:
+            if trial[1] == 'esriFieldTypeDate' and feature['attributes'][trial[0]] != None:
+                row.append(to_time(feature['attributes'][trial[0]]))
+            else:
+                row.append(feature['attributes'][trial[0]])
+
+        row.append(geometry)
+        
+        rows.append(row)
+    
+    
+    insert=sql.SQL("INSERT INTO {schema_table} ({columns}) VALUES %s").format(
+        schema_table = sql.Identifier(schema_name, output_table), 
+        columns = insert_column
+    )
+    with con:
+        with con.cursor() as cur:
+               execute_values(cur, insert, rows)
+    LOGGER.info('Successfully inserted %d records into %s', len(rows), output_table)
+
+
+def get_layer(mapserver_n, layer_id, schema_name, con = None):
```

Comment:
Can we reuse `get_layer` from gcc_layer_functions.py with some of the params fixed using `partial`?